composer/callbacks/callback_hparams.py

```diff
@@ -153,3 +155,59 @@ class TorchProfilerHparams(CallbackHparams):
     def initialize_object(self) -> TorchProfiler:
         from composer.callbacks.torch_profiler import TorchProfiler
         return TorchProfiler(**asdict(self))
+
+
+@dataclass
+class RunDirectoryUploaderHparams(CallbackHparams):
+    """:class:`~composer.callbacks.torch_profiler.RunDirectoryUploader` hyperparameters.
+
+    See :class:`~composer.callbacks.torch_profiler.RunDirectoryUploader` for documentation.
+    """
+
+    # Args:
+    #     provider_init_kwargs (Dict[str, Any], optional): Parameters to pass into the constructor for the
+    #         :class:`~libcloud.storage.providers.Provider` constructor. These arguments would usually include the cloud region
+    #         and credentials. Defaults to None, which is equivalent to an empty dictionary.
+    provider: str = hp.required("Cloud provider to use.")
+    container: str = hp.optional("he name of the container (i.e. bucket) to use.", default=None)
+    key: Optional[str] = hp.optional(textwrap.dedent(
+        """API key or username to use to connect to the provider. For security. do NOT hardcode the key in the YAML.
+        Instead, please specify via CLI arguments, or even better, environment variables."""),
+                                     default=None)
+    secret: Optional[str] = hp.optional(textwrap.dedent(
```

Comment:
Should probably just make this a file path rather than rely on the user to not hardcode the key - see #81 for an example

Reply:
Going to leave it as is for now. Would rather not add extra environment variables in everywhere. By January, YAHP will have better inheritance for secret files.